workitem/enum_type_blackbox_test.go

```diff
@@ -321,3 +321,86 @@ func TestEnumType_EqualEnclosing(t *testing.T) {
 		})
 	})
 }
+
+func TestEnumType_ConvertFromModel(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	type testCase struct {
+		subTestName    string
+		input          interface{} // contains valid and invalid values
+		expectedOutput interface{}
+		wantErr        bool
+	}
+	tests := []struct {
+		name string
+		enum w.EnumType
+		data []testCase
+	}{
+		{
+			"kind string",
```

Comment:
You're using an enum of type string, why did you call this test "kind string" then? I suggest to have more tests for list, enums and simple types and test every one of them individually.

Reply:
This is a test for enum of type string. I can rename `Kind String` to `type string`. Would that be enough?

Reply:
Never mind. I didn't see the context for this subtest.